src/checkbox.jsx

```diff
@@ -169,6 +169,12 @@ let Checkbox = React.createClass({
   _handleStateChange(newSwitched) {
     this.setState({switched: newSwitched});
   },
+  
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.checked !== this.state.switched) {
+      this.setState({switched: nextProps.checked});
+    }
```

Comment:
@spacenick could you rebase and make the above change so we could go ahead and merge this?